Windows/Debugger/CtrlDisAsmView.cpp

```diff
@@ -1377,3 +1542,50 @@ u32 CtrlDisAsmView::getInstructionSizeAt(u32 address)
 	u32 next  = manager.getNthNextAddress(start,1);
 	return next-address;
 }
+
+int regStringToMIPSGPReg(const char* name) {
+	std::string Name = name;
+
+	if (GPR_NAMES.find(Name) != GPR_NAMES.end()) {
+		if (Name == "v0") return MIPS_REG_V0;
+		if (Name == "v1") return MIPS_REG_V1;
+		if (Name == "a0") return MIPS_REG_A0;
+		if (Name == "a1") return MIPS_REG_A1;
+		if (Name == "a2") return MIPS_REG_A2;
+		if (Name == "a3") return MIPS_REG_A3;
+		if (Name == "a4") return MIPS_REG_A4;
+		if (Name == "a5") return MIPS_REG_A5;
+		if (Name == "t0") return MIPS_REG_T0;
+		if (Name == "t1") return MIPS_REG_T1;
+		if (Name == "t2") return MIPS_REG_T2;
+		if (Name == "t3") return MIPS_REG_T3;
+		if (Name == "t4") return MIPS_REG_T4;
+		if (Name == "t5") return MIPS_REG_T5;
+		if (Name == "t6") return MIPS_REG_T6;
+		if (Name == "t7") return MIPS_REG_T7;
+		if (Name == "t8") return MIPS_REG_T8;
+		if (Name == "t9") return MIPS_REG_T9;
+		if (Name == "s0") return MIPS_REG_S0;
+		if (Name == "s1") return MIPS_REG_S1;
+		if (Name == "s2") return MIPS_REG_S2;
+		if (Name == "s3") return MIPS_REG_S3;
+		if (Name == "s4") return MIPS_REG_S4;
+		if (Name == "s5") return MIPS_REG_S5;
+		if (Name == "s6") return MIPS_REG_S6;
+		if (Name == "s7") return MIPS_REG_S7;
+		if (Name == "k0") return MIPS_REG_K0;
+		if (Name == "k1") return MIPS_REG_K1;
+		if (Name == "gp") return MIPS_REG_GP;
+		if (Name == "sp") return MIPS_REG_SP;
+		if (Name == "fp") return MIPS_REG_FP;
+		if (Name == "ra") return MIPS_REG_RA;
+		if (Name == "hi") return MIPS_REG_HI;
+		if (Name == "lo") return MIPS_REG_LO;
+	}
+	else if (Name == "zero") {
+		return MIPS_REG_ZERO;
+	}
+	else return MIPS_REG_INVALID;
+	// extra return just to make the compiler happy.
+	return MIPS_REG_INVALID;
```

Comment:
It's not just to make the compiler happy, let's say someone add a new name to GPR_NAMES and forget to add it in this function, what will happen? I might sound silly now, but if you edit your code after some year is kinda easy for thing like this to happen.

I would remove the `else return MIPS_REG_INVALID;` like tho' as that is not needed and just left the last return if we didn't catch any condition before.

Reply:
Im confused what you are suggesting here? (though i did fix my code according to your suggestion upthread)